artiq/applets/plot_hist.py

```diff
@@ -10,24 +11,38 @@ class HistogramPlot(pyqtgraph.PlotWidget):
     def __init__(self, args):
         pyqtgraph.PlotWidget.__init__(self)
         self.args = args
+        self.timer = QTimer()
+        self.timer.timeout.connect(self.warning)
 
     def data_changed(self, data, mods, title):
         try:
-            y = data[self.args.y][1]
+            self.y = data[self.args.y][1]
             if self.args.x is None:
-                x = None
+                self.x = None
             else:
-                x = data[self.args.x][1]
+                self.x = data[self.args.x][1]
         except KeyError:
             return
-        if x is None:
-            x = list(range(len(y)+1))
+        if self.x is None:
+            self.x = list(range(len(self.y)+1))
 
-        if len(y) and len(x) == len(y) + 1:
+        if len(self.y) and len(self.x) == len(self.y) + 1:
             self.clear()
-            self.plot(x, y, stepMode=True, fillLevel=0,
+            self.plot(self.x, self.y, stepMode=True, fillLevel=0,
                       brush=(0, 0, 255, 150))
             self.setTitle(title)
+        else:
+            self.timer.start(1000)
+
+    def warning(self):
+        if len(self.y) and len(self.x) == len(self.y) + 1:
+            return
```

Comment:
What about stopping the timer in ``data_changed`` instead, so you don't have to store the data and duplicate this test?